service/kas/access/rewrap.go

```diff
@@ -318,27 +318,33 @@ func (p *Provider) tdf3Rewrap(ctx context.Context, body *RequestBody, entity *en
 
 	access, err := canAccess(ctx, ent, *policy, p.SDK)
 
+	auditPolicy := transformAuditPolicy(policy)
+
 	if err != nil {
-		slog.WarnContext(ctx, "Could not perform access decision!", "err", err)
+		p.Logger.WarnContext(ctx, "Could not perform access decision!", "err", err)
+		p.Logger.AuditRewrap(ctx, *auditPolicy, false)
```

Comment:
What kind of errors are returned? I do get that technically the key was not released but it does feel like audit on errors could be confusing 🤷

Reply:
This is not as much of an error as an error log. See this part of the Python implementation: https://github.com/opentdf/backend/blob/dbfcea7594fa72dd49d02a19aa01b36f3a32b7e6/containers/kas/kas_app/tdf3_kas_app/plugins/audit_hooks.py#L85